pybricksdev/cli/__init__.py

```diff
@@ -280,21 +281,38 @@ async def reconnect_hub():
                     response = await hub.race_disconnect(
                         hub.race_power_button_press(
                             questionary.select(
-                                "Would you like to re-compile your code?",
+                                f"Would you like to re-compile {os.path.basename(args.file.name)}?",
                                 response_options,
                                 default=(
-                                    response_options[0]
+                                    response_options[1]
                                     if args.start
-                                    else response_options[1]
+                                    else response_options[2]
                                 ),
                             ).ask_async()
                         )
                     )
+
+                    if response == response_options[0]:
+                        args.file.close()
+                        args.file = open(
```

Comment:
It wouldn't hurt to handle an error if the file doesn't exist. If the user types in the file name wrong, it would be nice to give another chance instead of crashing.